server/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -1485,6 +1536,19 @@ private SearchSourceBuilder parseXContent(XContentParser parser, boolean checkTr
                         }
                     }
                     searchUsage.trackSectionUsage(KNN_FIELD.getPreferredName());
+                } else if (QUERIES_FIELD.match(currentFieldName, parser.getDeprecationHandler())) {
```

Comment:
It it ok to have only a single query object under `queries`?

Reply:
I think it makes sense to allow that if you wanted to sort on the individual query (future improvement), but not on a knn search as part of ranking.

Reply:
Since we are modifying this file, should we also make sure that (de)/serialization  and from/toXcontent tests are covered in `SearchSourceBuilderTests` by adding `queries` part to `RandomSearchRequestGenerator`?

Reply:
Yes, I will add this there.

Reply:
I added this.